src/cpu_features/aarch64.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+use arg_enum_proc_macro::ArgEnum;
+use std::env;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, ArgEnum)]
+pub enum CpuFeatureLevel {
+  NATIVE,
+  NEON,
+}
+
+impl CpuFeatureLevel {
+  pub const fn len() -> usize {
+    CpuFeatureLevel::NEON as usize + 1
+  }
+
+  #[inline(always)]
+  pub fn as_index(self) -> usize {
+    const LEN: usize = CpuFeatureLevel::len();
+    assert_eq!(LEN & (LEN - 1), 0);
+    self as usize & (LEN - 1)
+  }
+}
+
+impl Default for CpuFeatureLevel {
+  fn default() -> CpuFeatureLevel {
+    let detected = CpuFeatureLevel::NEON;
```

Comment:
You would need to use [is_aarch64_feature_detected](https://doc.rust-lang.org/std/macro.is_aarch64_feature_detected.html).

And/or you have to document either that aarch64 needs nightly OR that aarch64 by default requires neon.

Reply:
We can elide `is_aarch64_feature_detected!()` here because aarch64 mandates NEON. We maintain `CpuFeatureLevel` primarily to force native Rust kernels at runtime, in a way consistent with other architectures.
We should include a comment to that effect for future readers.

Reply:
@barrbrain advised to have default requires NEON.
 I will document about it so there won't be any confusion

Reply:
We imply already sse2 in x68_64 and it is fine to require neon, but I would document it so people with peculiar setups or cores aren't surprised.

On upstream rust/stdarch I already got some requests to not assume altivec+vsx on the power8/power9le targets since there are few cores that do not have it...